src/ast/passes/macro_expansion.cpp

```diff
@@ -198,6 +205,11 @@ std::optional<Block *> MacroExpander::expand(Macro &macro, const Call &call)
                                      Location(call.loc)),
             clone(ast_, call.vargs.at(i), call.vargs.at(i).loc()),
             Location(call.loc)));
+        // As per the name these arg variables are not allowed to be mutated
+        // because the caller passed in an expression instead of another
+        // variable e.g.
+        // macro add1($x) { $x += 1; $x } BEGIN { add1(1 + 1);
+        arg_vars_no_mutation.insert(mvar->ident);
```

Comment:
Can't we reuse the value inserted into `temp_vars_` below?

Reply:
No. `temp_vars_` holds the updated name via `get_new_var_ident`, we need the original one in the macro args.

Reply:
Oh, right, now I get it. Continuing the discussion in another thread for better visibility.